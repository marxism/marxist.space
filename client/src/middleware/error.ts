import {
  FETCH_RESOURCES_SUCCESS,
  FETCH_RESOURCES_ERROR,
  FETCH_RESOURCE_SUCCESS,
  FETCH_RESOURCE_ERROR,
} from '../actions/resources'
import { hideErrorClearTimeout, showErrorWithTimeout } from '../actions/error'

export default (store) => (next) => (action) => {
  next(action)
  switch (action.type) {
    case FETCH_RESOURCES_SUCCESS:
    case FETCH_RESOURCE_SUCCESS:
      if (store.getState().error) {
        store.dispatch(hideErrorClearTimeout())
      }
      break

    case FETCH_RESOURCES_ERROR:
    case FETCH_RESOURCE_ERROR:
      store.dispatch(showErrorWithTimeout(action.error))
      break

    default:
      break
  }
}
