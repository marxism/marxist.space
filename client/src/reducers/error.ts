import {
  FETCH_RESOURCES_ERROR,
  SEARCH_RESOURCES_ERROR,
  FILTER_RESOURCES_ERROR,
} from '../actions/resources'
import { HIDE_ERROR } from '../actions/error'

const initialState = null

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_RESOURCES_ERROR:
    case FILTER_RESOURCES_ERROR:
    case SEARCH_RESOURCES_ERROR:
    case HIDE_ERROR:
      return null

    default:
      return state
  }
}
