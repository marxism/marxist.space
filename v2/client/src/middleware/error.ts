import {
  FETCH_POSTS_SUCCESS,
  FETCH_POSTS_ERROR,
  FETCH_POST_SUCCESS,
  FETCH_POST_ERROR,
} from '../actions/posts'
import { hideErrorClearTimeout, showErrorWithTimeout } from '../actions/error'

export default (store) => (next) => (action) => {
  next(action)
  switch (action.type) {
    case FETCH_POSTS_SUCCESS:
    case FETCH_POST_SUCCESS:
      if (store.getState().error) {
        store.dispatch(hideErrorClearTimeout())
      }
      break

    case FETCH_POSTS_ERROR:
    case FETCH_POST_ERROR:
      store.dispatch(showErrorWithTimeout(action.error))
      break

    default:
      break
  }
}
