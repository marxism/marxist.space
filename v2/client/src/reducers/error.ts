import {
  FETCH_POSTS_ERROR,
  FETCH_POST_ERROR,
} from '../actions/posts'
import { HIDE_ERROR } from '../actions/error'

const initialState = null

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_POSTS_ERROR:
    case FETCH_POST_ERROR:
    case HIDE_ERROR:
      return null

    default:
      return state
  }
}
