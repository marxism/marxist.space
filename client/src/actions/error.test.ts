/* eslint-env jest */

import * as errorActions from './error'

describe('error actions', () => {
  test('showError', () => {
    expect(errorActions.showError('oh no')).toMatchSnapshot()
  })

  test('hideError', () => {
    expect(errorActions.hideError()).toMatchSnapshot()
  })
})
