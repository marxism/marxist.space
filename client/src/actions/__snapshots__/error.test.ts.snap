// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error actions hideError 1`] = `
Object {
  "type": "HIDE_ERROR",
}
`

exports[`error actions showError 1`] = `
Object {
  "error": "oh no",
  "type": "SHOW_ERROR",
}
`
